opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,513 @@
+
+    def __init__(self,
+                 value: str) -> None:
+        self.value = value
+
+
+class MetricOptions:
+    def __init__(self,
+                 description: str,
+                 unit: str,
+                 label_keys: typing.List['LabelKey'],
+                 constant_labels: typing.Dict['LabelKey', 'LabelValue'],
+                 component: str,
+                 resource: 'Resource') -> None:
+        """Optional info used when creating a :class:`.Metric`.
+
+        Args:
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values.
+            Order of the list is important as the same order MUST be used
+            on recording when suppling values for these labels
+            constant_labels: A map of constant labels that will be used for
+            all of the TimeSeries created from the Metric.
+            component: The name of the component that reports this metric.
+            Resource: Sets the :class:`.Resource` associated with this
+            Metric.
+        """
+        self.description = description
+        self.unit = unit
+        self.label_keys = label_keys
+        self.constant_labels = constant_labels
+        self.component = component
+        self.resource = resource
+
+
+class MeasureType:
+    DOUBLE = 0
+    LONG = 1
+
+
+class MeasureOptions:
+    def __init__(self,
+                 description: str,
+                 unit: str,
+                 measure_type: 'MeasureType' = MeasureType.DOUBLE
+                 ) -> None:
+        """Optional info used when creating a :class:`.Measure`.
+
+        Args:
+            description: Human readable description of this measure.
+            unit: Unit of the measure values.
+            measure_type: Type of the measure. Can be one of two values -
+            `LONG` and `DOUBLE`. Default type is `DOUBLE`.
+        """
+        self.description = description
+        self.unit = unit
+        self.measure_type = measure_type
+
+
+class RecordOptions:
+
+    def __init__(self,
+                 distributed_context: 'DistributedContext',
+                 span_context: 'SpanContext') -> None:
+        """Optional info used when recording :class:`.Measurement`s.
+
+        Args:
+            distributed_context: Explicit :class:`.DistributedContext` to use
+            instead of the current context. Context is used to add dimensions
+            for the resulting metric calculated out of the provided
+            measurements.
+            span_context: the :class:`.SpanContext` that identified the
+            :class:`.Span` for which the measurements are associated with.
+        """
+
+
+class Measurement:
+    """An empty interface that represents a single value.
+
+    This single value is recorded for the :class:`.Measure` that created
+    this measurement.
+    """
+
+
+class Measure:
+
+    def __init__(self,
```

Comment:
What's the reason for putting the constructor in the API? This seems like an implementation detail since the API includes `Meter.create_measure`.